include/aws/mqtt/private/v5/mqtt3_to_mqtt5_adapter_impl.h

```diff
@@ -78,6 +94,46 @@ struct aws_mqtt3_to_mqtt5_adapter_operation_publish {
     void *on_publish_complete_user_data;
 };
 
+struct aws_mqtt3_to_mqtt5_adapter_operation_subscribe {
+    struct aws_mqtt3_to_mqtt5_adapter_operation_base base;
+
+    /*
+     * holds a reference to the MQTT5 client subscribe operation until the operation completes or our adapter
+     * goes away.
+     *
+     * In the case where we're going away, we zero out the MQTT5 operation callbacks to prevent crash-triggering
+     * notifications.
+     */
+    struct aws_mqtt5_operation_subscribe *subscribe_op;
+
+    /* aws_array_list<struct aws_mqtt_subscription_set_subscription_record *> */
```

Comment:
delete?

Reply:
It's communicating what the container type is since we don't have generics in C